public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.test.ts

```diff
@@ -0,0 +1,290 @@
+import { M } from 'msw/lib/glossary-dc3fd077';
+
+import LokiLanguageProvider from '../../../LanguageProvider';
+import { LokiDatasource } from '../../../datasource';
+import { createLokiDatasource } from '../../../mocks';
+
+import { CompletionDataProvider } from './CompletionDataProvider';
+import { getCompletions } from './completions';
+import { Label, Situation } from './situation';
+
+const history = [
+  {
+    ts: 12345678,
+    query: {
+      refId: 'test-1',
+      expr: '{test: unit}',
+    },
+  },
+  {
+    ts: 87654321,
+    query: {
+      refId: 'test-1',
+      expr: '{test: unit}',
+    },
+  },
+];
+
+const labelNames = ['place', 'source'];
+const labelValues = ['moon', 'luna'];
+const extractedLabelKeys = ['extracted', 'label'];
+const otherLabels: Label[] = [
+  {
+    name: 'place',
+    value: 'luna',
+    op: '=',
+  },
+];
+const afterSelectorCompletions = [
+  {
+    insertText: '|= "$0"',
+    isSnippet: true,
+    label: '|= "something"',
+    type: 'LINE_FILTER',
+  },
+  {
+    insertText: '!= "$0"',
+    isSnippet: true,
+    label: '!= "something"',
+    type: 'LINE_FILTER',
+  },
+  {
+    insertText: '|~ "$0"',
+    isSnippet: true,
+    label: '|~ "something"',
+    type: 'LINE_FILTER',
+  },
+  {
+    insertText: '!~ "$0"',
+    isSnippet: true,
+    label: '!~ "something"',
+    type: 'LINE_FILTER',
+  },
+  {
+    insertText: '',
+    label: '// Placeholder for the detected parser',
+    type: 'DETECTED_PARSER_PLACEHOLDER',
+  },
+  {
+    insertText: '',
+    label: '// Placeholder for logfmt or json',
+    type: 'OPPOSITE_PARSER_PLACEHOLDER',
+  },
+  {
+    insertText: 'pattern',
+    label: 'pattern',
+    type: 'PARSER',
+  },
+  {
+    insertText: 'regexp',
+    label: 'regexp',
+    type: 'PARSER',
+  },
+  {
+    insertText: 'unpack',
+    label: 'unpack',
+    type: 'PARSER',
+  },
+  {
+    insertText: 'unwrap extracted',
```

Comment:
Why is unwrap `LINE_FILTER`. It is not line filter. It is kind of category of its own. So I would either create new category or create `PIPE_OPERATIONS` and add there all operations after | (besides parsers). 🤔

Reply:
Could you please give me more context to understand this? Is there any reference I can use for this? Thanks!

Reply:
Here is the documentation https://grafana.com/docs/loki/latest/logql/log_queries and here is a miro board I once created for this project: https://miro.com/app/board/uXjVOF5z1u8=/. 

The issue here is that unwrap is pipe operation in metrics query and it doesn't belong to any category mentioned in log queries.Here are categories for log operations: 
<img width="665" alt="image" src="https://user-images.githubusercontent.com/30407135/194115866-dd7b87ac-d9d7-444e-9afd-e355e26baf22.png">

[Here is the documentation for unwrap](https://grafana.com/docs/loki/latest/logql/metric_queries/#unwrapped-range-aggregations) and it called `unwrap expression`.

So either `unwrap` should have category on its own in the same way as you created category for `label_format` and `line_format`, which I think would be preferable. Or we could create `PIPE_OPERATIONS` group that would include `line_format`, `label_format` and `unwrap.`